crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -39,54 +39,47 @@ fn toml_map_to_field(
 ) -> Result<BTreeMap<String, InputValue>, InputParserError> {
     let mut field_map = BTreeMap::new();
     for (parameter, value) in toml_map {
-        match value {
+        let mapped_value = match value {
             TomlTypes::String(string) => {
                 let new_value = parse_str(&string)?;
+
                 if new_value.is_none() {
-                    check_toml_map_duplicates(&mut field_map, parameter, InputValue::Undefined)?
+                    InputValue::Undefined
                 } else {
-                    check_toml_map_duplicates(
-                        &mut field_map,
-                        parameter,
-                        InputValue::Field(new_value.unwrap()),
-                    )?
+                    InputValue::Field(new_value.unwrap())
                 }
             }
             TomlTypes::Integer(integer) => {
                 let new_value = parse_str(&integer.to_string())?;
-                check_toml_map_duplicates(
-                    &mut field_map,
-                    parameter,
-                    InputValue::Field(new_value.unwrap()),
-                )?
+
+                InputValue::Field(new_value.unwrap())
             }
             TomlTypes::Bool(boolean) => {
                 let new_value = if boolean { FieldElement::one() } else { FieldElement::zero() };
-                check_toml_map_duplicates(&mut field_map, parameter, InputValue::Field(new_value))?
+
+                InputValue::Field(new_value)
             }
             TomlTypes::ArrayNum(arr_num) => {
                 let array_elements: Vec<_> = arr_num
                     .into_iter()
                     .map(|elem_num| parse_str(&elem_num.to_string()).unwrap().unwrap())
                     .collect();
-                check_toml_map_duplicates(
-                    &mut field_map,
-                    parameter,
-                    InputValue::Vec(array_elements),
-                )?
+
+                InputValue::Vec(array_elements)
             }
             TomlTypes::ArrayString(arr_str) => {
                 let array_elements: Vec<_> = arr_str
                     .into_iter()
                     .map(|elem_str| parse_str(&elem_str).unwrap().unwrap())
                     .collect();
-                check_toml_map_duplicates(
-                    &mut field_map,
-                    parameter,
-                    InputValue::Vec(array_elements),
-                )?
+
+                InputValue::Vec(array_elements)
             }
-        }
+        };
+
+        if field_map.insert(parameter.clone(), mapped_value).is_some() {
+            return Err(InputParserError::DuplicateVariableName(parameter));
+        };
```

Comment:
This check on the return value seems a little odd and could be removed. `BTreeMap` not having multiple values for the same key is guaranteed but I'm also not too concerned about this.